fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/mixin/object/builder/ModelPredicateProviderRegistrySpecificAccessor.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.object.builder;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.gen.Invoker;
+
+import net.minecraft.client.item.ModelPredicateProvider;
+import net.minecraft.client.item.ModelPredicateProviderRegistry;
+import net.minecraft.item.Item;
+import net.minecraft.util.Identifier;
+
+// mixin doesn't care about descriptor, must put two "register" accessors in different places
```

Comment:
If Mixin is ignoring the descriptors then why does putting these in two different classes solve the problem?

Reply:
Because refmap entries in each class are stored in separate subobjects like https://gist.github.com/46d619f045e5da96ffa28a9d20366d5d
so when the two `register`s are in two different classes they won't clash as they are in different json objects

Reply:
makes sense. that's a bit stupid of the refmap system